gtsfm/ui/dot_graph_generator.py

```diff
@@ -0,0 +1,137 @@
+"""
+Creates a DOT graph based on the contents of RegistryHolder.REGISTRY.
+
+Author: Kevin Fu
+"""
+
+from gtsfm.ui.registry import RegistryHolder
+
+from pathlib import Path
+import os
+
+import pydot
+from collections import namedtuple
+
+from typing import Tuple
+from gtsfm.ui.registry import UiMetadata
+
+JS_ROOT = os.path.join(Path(__file__).resolve().parent.parent.parent, "rtf_vis_tool")
+
+
+class DotGraphGenerator:
+    """Generates and saves a graph of all the components in REGISTRY."""
+
+    def __init__(self, test_mode=False):
+        """Create DotGraphGenerator.
+
+        Args:
+            test_mode - boolean flag, only set True when unit testing.
```

Comment:
Per the [style guide](https://github.com/borglab/gtsfm/blob/master/CONTRIBUTING.md), let's update this to colon separated. Thanks!